app/models/sample.rb

```diff
@@ -25,11 +25,41 @@ class Sample < ApplicationRecord
   include BasespaceHelper
   include ErrorHelper
 
+  belongs_to :project
+  # This is the user who uploaded the sample, possibly distinct from the user(s) owning the sample's project
+  belongs_to :user, counter_cache: true # use .size for cache, use .count to force COUNT query
+  belongs_to :host_genome, counter_cache: true # use .size for cache, use .count to force COUNT query
+  has_many :pipeline_runs, -> { order(created_at: :desc) }, dependent: :destroy
+  has_many :backgrounds, through: :pipeline_runs
+  has_many :input_files, dependent: :destroy
+  accepts_nested_attributes_for :input_files
+  validates_associated :input_files
+  has_many :metadata, dependent: :destroy
+  has_and_belongs_to_many :visualizations
+
   STATUS_CREATED = 'created'.freeze
   STATUS_UPLOADED = 'uploaded'.freeze
   STATUS_RERUN    = 'need_rerun'.freeze
   STATUS_RETRY_PR = 'retry_pr'.freeze # retry existing pipeline run
   STATUS_CHECKED = 'checked'.freeze # status regarding pipeline kickoff is checked
+  validates :status, presence: true, inclusion: { in: [
+    STATUS_CREATED,
+    STATUS_UPLOADED,
+    STATUS_RERUN,
+    STATUS_RETRY_PR,
+    STATUS_CHECKED,
+  ], }, if: :mass_validation_enabled?
+
+  validate :input_files_checks
+  validates :name, presence: true, uniqueness: { scope: :project_id, case_sensitive: false }
```

Comment:
Should we use `, if: :mass_validation_enabled?` here as well?

Reply:
This one actually already existed as-is. I just moved it.